pkg/build/builder/docker.go

```diff
@@ -69,12 +70,22 @@ func (d *DockerBuilder) Build() error {
 	}
 	sourceInfo, err := fetchSource(d.dockerClient, buildDir, d.build, d.urlTimeout, os.Stdin, d.gitClient)
 	if err != nil {
+		d.build.Status.Reason = api.StatusReasonFetchSourceFailed
+		d.build.Status.Message = api.StatusMessageFetchSourceFailed
+		if updateErr := updateStatusRetrier(d.build, d.client, nil); updateErr != nil {
+			glog.V(0).Infof("error: An error occured while updating the build status: %v", updateErr)
+		}
 		return err
 	}
+
 	if sourceInfo != nil {
-		updateBuildRevision(d.client, d.build, sourceInfo)
+		glog.V(4).Infof("Setting build revision with details %#v", sourceInfo)
+		revision := updateBuildRevision(d.build, sourceInfo)
+		if updateErr := updateStatusRetrier(d.build, d.client, revision); updateErr != nil {
+			glog.V(0).Infof("error: An error occured while updating the build status: %v", updateErr)
```

Comment:
Never eat errors directly.  Write errors using `pkg/util/runtime.HandleError`.  Specifically handle the forbidden error by writing that to glog.V(2).Infof, send everything else to HandleError.